packages/mattermost-redux/src/selectors/entities/threads.ts

```diff
@@ -57,15 +56,13 @@ export function getThread(state: GlobalState, threadId: $ID<UserThread> | undefi
     return getThreads(state)[threadId];
 }
 
-export function getThreadOrSynthetic(state: GlobalState, threadId: $ID<UserThread>): UserThread | UserThreadSynthetic {
-    const thread = getThreads(state)[threadId];
+export function getThreadOrSynthetic(state: GlobalState, post: Post): UserThread | UserThreadSynthetic {
```

Comment:
> actually it should be (state, rootPost), else we are going to have incorrect data e.g reply_count is 0 for replies.

I agree, 100% we should be explicit `(state, rootPost)`

> We could throw if there is a post.root_id to enforce it, a bit HC though, what do you think?

I could see throwing if there's a `root_id` as a development env error, but I think it'd be overkill TBH.